### ISSUE TITLE
Crash when using field dressing

### ISSUE THREAD

**Issue by TiloWiklund** [CleverRaven/Cataclysm-DDA#24918]
**Describe the bug**  

The game crashes with "Error: attempt to dereference a past-the-end iterator" whenever I try to use field dressing.

**To Reproduce**  
1. find a corpse
2. apply field dressing

**Expected behavior**  

Not crashing

**Versions and configuration(please complete the following information):**
 - OS: Gentoo Linux
 - Game Version: 0.C-30671-gcb0c80d643 (commit cb0c80d643b29f6474cc29cbe53232fc70d41c4b)
 - Terminal
 - DDA + Filthy Clothing + Disable NPC Needs + Simplified Nutrition

**Additional context**  

I did a little bit of debugging, I'm not familiar with the source code but the following may be useful:

The crash happens because MonsterGenerator.flag_map not containing the key "FIELD_DRESS" in `MonsterGenerator::m_flag_from_string` (monstergenerator.cpp:733), so that trying to dereference the result from flag_map.find(flag) fails.

The original call is 
`corpse->has_flag( "FIELD_DRESS" )`
in line 784--786 in activity_handlers.cpp
```
bool make_offal = !chunk.is_tainted() && !chunk.has_flag( "CANNIBALISM" ) &&
                  !( corpse->has_flag( "FIELD_DRESS" ) || corpse->has_flag( "FIELD_DRESS_FAILED" ) ) &&
                  !chunk.made_of ( material_id ( "veggy" ) );

```

Relevant part of backtrace:
* `MonsterGenerator::m_flag_from_string` (monstergenerator.cpp:733) (flag = "FIELD_DRESS")
* `mtype::has_flag` (mtype.cpp:67)
* `butchery_drops_hardcoded` (activity_handlers.cpp:785)
* `activity_handlers::butcher_finish` (activity_handlers.cpp:1048)

**Comment by TiloWiklund** [comment]
Adding the flags [FIELD_DRESS, FIELD_DRESS_FAILED and QUARTERED](https://pastebin.com/7MCvvyu5) stops things from crashing and (at least at first sight) appears to give the correct behaviour when field dressing/(full) butchering/quartering.

**Comment by alanbrady** [comment]
You're right it is indeed broken.  It looks like MF flags weren't added to the bitops so it's going beyond MF_MAX in has_flag.

Working on a fix, will do PR after making sure it fixes it.

**Comment by alanbrady** [comment]
Oh nevermind, https://github.com/CleverRaven/Cataclysm-DDA/pull/24923 already fixes and I was actually wrong anyway.

**Comment by Night-Pryanik** [comment]
Closed by #24923.
